source/base/parameter_handler.cc

```diff
@@ -1936,11 +1936,28 @@ void ParameterHandler::parse_input_from_xml (std::istream &in)
 
   const std::size_t n_top_level_elements = std::distance (single_node_tree.begin(),
                                                           single_node_tree.end());
-  AssertThrow (n_top_level_elements == 1,
-               ExcInvalidXMLParameterFile ("There are "
-                                           + Utilities::to_string(n_top_level_elements)
-                                           + " top-level elements, but there "
-                                           "should only be one."));
+  if (n_top_level_elements != 1)
+    {
+      std::ostringstream top_level_message;
+      top_level_message << "There are "
+                        << n_top_level_elements
+                        << " top level elements, but there should only be one.\n"
+                        << "The top level elements are:\n";
```

Comment:
I think the error message would be easier to understand if you said somewhere something of the form "While reading a ParameterHandler input file in XML format, ..."